src/server/bg_services/server_monitor.js

```diff
@@ -503,5 +534,23 @@ async function _verify_and_update_public_ip() {
         });
 }
 
+async function _check_k8s_services_address_changes() {
+    dbg.log2('_check_k8s_services_address_changes');
+    try {
+        const system_id = system_store.data.systems[0]._id;
+        const system_address = await os_utils.discover_k8s_services();
+        await system_store.make_changes({
```

Comment:
Why do you need to call this make_changes() unconditionally?
The thing is that make_changes also needs to publish the changes and if there is no change to make we better avoid calling it.
Can you check if you really need to update?